src/webrtc/peerconnection.ts

```diff
@@ -381,21 +389,25 @@ export class PeerConnectionClass implements PeerConnection<signals.Message> {
     //       is guaranteed to be unique for 2 peers.
     switch(message.type) {
       case signals.Type.OFFER:
-        this.handleOfferSignalMessage_(message.description);
+        this.handleOfferSignalMessage_(message.description).then(
+            this.fulfillHaveRemoteDescription_);
         break;
       // Answer to an offer we sent
       case signals.Type.ANSWER:
-        this.handleAnswerSignalMessage_(message.description);
+        this.handleAnswerSignalMessage_(message.description).then(
+            this.fulfillHaveRemoteDescription_);
         break;
       // Add remote ice candidate.
       case signals.Type.CANDIDATE:
-        this.handleCandidateSignalMessage_(message.candidate);
+        this.onceHaveRemoteDescription_.then(() => {
+          this.handleCandidateSignalMessage_(message.candidate);
+        });
```

Comment:
Right, but `onceHaveRemoteDescription` never throws (we never call the rejection function).